Kernel/Arch/aarch64/RPi/AUX.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Singleton.h>
+#include <Kernel/Arch/aarch64/RPi/AUX.h>
+#include <Kernel/Arch/aarch64/RPi/MMIO.h>
+
+namespace Kernel::RPi {
+
+// bcm2711-peripherals.pdf "2. Auxiliaries: UART1, SPI1 & SPI2"
+struct AUXRegisters {
+    u32 IRQ;
+    struct {
+        u32 mini_uart_enable : 1;
```

Comment:
Are we OK using bit fields here? I know this code is specific to ARM64, just wondered if we are OK with using bit-fields as they're kinda useful in some cases but obviously a landmine in other situations (maybe not the case here...).

Reply:
Bit fields make this code look a bit nicer IMO. Without bit fields we would have to manually clear/set bits with `&= ~`/`|=`.

Bit fields should be safe to use here, since this code is only ever compiled on aarch64 (and we will hopefully never support big endian aarch64). This struct also isn't packed, so we won't have any weird alignment problems here.

Reply:
Seems reasonable to me :)